src/Gitlab.purs

```
@@ -0,0 +1,76 @@
+module Gitlab where
+
+import Prelude
+
+import Control.Monad.Aff (Aff)
+import Data.Either (Either(..))
+import Data.Maybe (Maybe(..))
+import Network.HTTP.Affjax (AJAX, get)
+import Simple.JSON (readJSON)
+
+type BaseUrl = String
+type Token = String
+
+type Project =
+  { id   :: Int
+  , name :: String
+  }
+
+type User =
+  { avatar_url :: String
+  }
+
+type Commit =
+  { title    :: String
+  , short_id :: String
+  }
+
+type Pipeline =
+  { id     :: Int
+  , status :: String -- TODO: make enum?
+  }
+
+type Job =
+  { project     :: Maybe Project
+  , user        :: User
+  , commit      :: Commit
+  , ref         :: String
+  , pipeline    :: Pipeline
+  , status      :: String -- TODO: make enum
+  , created_at  :: String
+  , started_at  :: Maybe String
+  , finished_at :: Maybe String
+  }
+
+type Projects = Array Project
+type Jobs = Array Job
+
+
+getProjects :: forall a. BaseUrl -> Token -> Aff (ajax :: AJAX | a) Projects
+getProjects baseUrl token = do
+  let url = baseUrl
+            <> "/api/v4/projects?private_token="
+            <> token
+            <> "&simple=true&per_page=20&order_by=last_activity_at"
+  projectsRes <- get url
+  let ps = case readJSON projectsRes.response of
+        Left e -> []
```

Please, let's not swallow the errors in such way.

We are working inside `Aff`, error handling (throwing/catching) is a core feature of `Aff` (and it got very cool operations for that, like `apothize`, `attempt`, `invincible`, `supervise`, `bracket`). So throwing errors is morally right thing to do. `get` can throw in a bunch of circumstances (like transport errors), throwing on deserialization failure which can't be handled in any good way and renders the application empty goes very much in line with that.

They document it quite well [here](https://github.com/slamdata/purescript-aff#2-monaderror), just `throwError`.